### ISSUE TITLE
manual transition to unknown state still transitions

### ISSUE THREAD

**Issue by guihomework** [ubi-agni/pyhsm#12]
since issues existed in the viewer, I thought I would use the command line to request a transition in my state machine so did this
```
rostopic pub  /hsm_introspection/transition std_msgs/String "data: 'pick_place_hsm/root/PickPlace/wait'"
```
which resulted in this display 
```
exiting done
exiting idle
entering PickPlace
entering wait
```
all fine

now if I send any name to the transition topic, or even empty,
```
rostopic pub  /hsm_introspection/transition std_msgs/String "data: 'foo'"
```
 it will transition to idle which is my initial state.
```
exiting wait
exiting PickPlace
entering idle
Clearing planning scene
entering done
```

is that expected ?

**Comment by rhaschke** [comment]
"Your" fault: The state name needs to have the root state name at least. We can discuss, whether we should catch this error...

**Comment by guihomework** [comment]
Why is this closed ?  It is not working for me, even with the 'root' in the state. 

transition from idle to PickPlace/wait using 
``` 
rostopic pub  /hsm_introspection/transition std_msgs/String "data: 'root/PickPlace/wait'" 
```
does a wrong re-enter of idle state
```
exiting done
exiting idle
entering idle
Clearing planning scene
entering done
```

transition from idle to PickPlace/wait with 
```
rostopic pub  /hsm_introspection/transition std_msgs/String "data: 'pick_place_hsm/root/PickPlace/wait'"
```

works
```
exiting done
exiting idle
entering PickPlace
entering wait
```

and finally trying the same with a starting slash does absolutely nothing, which I prefer if it is a wrong syntax.
```
rostopic pub  /hsm_introspection/transition std_msgs/String "data: '/root/PickPlace/wait'"
```

Could you try to reproduce with the correct value (it is undocumented so I cannot guess, I used current_state to guess the syntax)  you think should do the transition correctly ?

Anything that does not correspond should not transition to my opinion.

**Comment by rhaschke** [comment]
Sorry, indeed there were two other bugs not correctly checking for invalid state names.
Should be fixed with 2b735ac.
